x-pack/plugins/enterprise_search/public/applications/app_search/components/engine/engine_logic.ts

```diff
@@ -87,6 +90,14 @@ export const EngineLogic = kea<MakeLogicType<EngineValues, EngineActions>>({
       () => [selectors.engine],
       (engine) => engine?.unconfirmedFields?.length > 0,
     ],
+    searchKey: [
+      () => [selectors.engine],
+      (engine: Partial<EngineDetails>) => {
+        const isSearchKey = (token: ApiToken) => token.type === ApiTokenTypes.Search;
```

Comment:
You could pull this function out instead of declaring it in here every time (idk if its worth writing a test for it)

Reply:
Going completely the opposite direction, I would have just left it as an inline fn in `.find()` :trollface: But it doesn't bother me either way personally

Reply:
Haha, quite the array of opinions! I just pulled this out for readability. It was long in the tooth inline in the `find`.